packages/pieces/src/apps/mailchimp/actions/add-member-to-list.action/add-member-to-list.action.ts

```diff
@@ -0,0 +1,111 @@
+            required: true,
+        }),
+        listId: Property.Dropdown<string>({
+            displayName: "Audience",
+            refreshers: ["authentication"],
+            description: "Audience you want to add the contact to",
+            required: true,
+            options: async (propsValue) => {
+                if (propsValue['authentication'] === undefined) {
+                    return {
+                        disabled: true,
+                        options: [],
+                        placeholder: "Please select an authentication"
+                    }
+                }
+                const authProp: AuthPropertyValue = propsValue['authentication'] as AuthPropertyValue;
+                let lists =  (await getUserLists(authProp)).lists;
+                return {
+                    disabled: false,
+                    options: lists.map(l => {
+                        return {
+                            label:l.name,
+                            value: l.id
+                        }
+                    })
+                };
+            }
+        }),
+        status: Property.Dropdown<'subscribed' | 'unsubscribed' | 'cleaned' | 'pending' | 'transactional'>({
+            displayName: "Status",
+            refreshers: [],
+            required: true,
+            options: async () => {
+            return {disabled:false, options:[
+                {label:'Subscribed', value:'subscribed'},
+                {label:'Unsubscribed', value: 'unsubscribed'},
+                {label:'Cleaned', value:'cleaned'},
+                {label: 'Pending', value: 'pending'},
+                {label:'Transactional',value:'transactional'}
+            ]} ;}
+            
+        })
+    },
+    
+    async run(context) {
+        const access_token= context.propsValue.authentication?.access_token;
+        const mailChimpMetaDataRequest: HttpRequest<{dc:string}> = {
+            method: HttpMethod.GET,
+            url: 'https://login.mailchimp.com/oauth2/metadata',
+            headers: {
+                Authorization: `OAuth ${access_token}`
+              }
+        };
+        const mailChimpServerPrefix = (await httpClient.sendRequest(mailChimpMetaDataRequest)).dc;
+        mailchimp.setConfig({
+            accessToken: access_token,
+            server: mailChimpServerPrefix
+          });
+        
+       return await mailchimp.lists.addListMember(context.propsValue.listId!,{email_address:context.propsValue.email!, status:context.propsValue.status!})
+    },
+});
+async function getUserLists(authProp: AuthPropertyValue): Promise<{lists:MailChimpList[]}> {
+    const access_token= authProp.access_token;
+    console.log(authProp);
+        const mailChimpMetaDataRequest: HttpRequest<{dc:string}> = {
+            method: HttpMethod.GET,
+            url: 'https://login.mailchimp.com/oauth2/metadata',
+            headers: {
+                Authorization: `OAuth ${access_token}`
+              }
+        };
```

Comment:
This can be moved to an common helper function.

Reply:
done